### DISCUSSION TITLE
S3 for storage?

### DISCUSSION THREAD

**Discussion by gnydick** [zinclabs/zinc#155] [General]
Is it just me or does it seem odd to use S3 to store indexes?

**Comment by prabhatsharma** [comment]
Why do you think it would be odd to use s3 for storage? 

EBS volume cost 8 cents/GB, then you create 2 more copies of it in a cluster. This results in 8x3=24 cents/GB cost. Using s3 allows you to store data at 2.3 cents/GB (10x lower cost). If your use case requires high throughput and is not latency sensitive then s3 is perfect choice.

**Comment by gnydick** [comment]
Because you can't use S3 like a block device. Isn't it incredibly slow?

**Reply by prabhatsharma** [reply]
speed can be defined as a combination of latency (time to first byte) and throughput (Bytes/second).  

On latency front s3 is going to be considerably slower but still under 200 ms mark for most calls. This could be very high latency for some applications like search implementation in websites. However it can be good enough for applications involving log search. 

Throughput between ec2 and s3 can be as high as 100Gbps. This is great throughput for most applications.
